src/Integration.Vsix/Settings/GeneralOptionsDialogControl.xaml.cs

```diff
@@ -30,29 +29,41 @@ namespace SonarLint.VisualStudio.Integration.Vsix
     /// </summary>
     public partial class GeneralOptionsDialogControl : UserControl
     {
-        private ISonarLintSettings settings;
-        private ISonarLintDaemon daemon;
+        private readonly ISonarLintSettings settings;
+        private readonly ISonarLintDaemon daemon;
 
-        public GeneralOptionsDialogControl()
+        public GeneralOptionsDialogControl(ISonarLintSettings settings, ISonarLintDaemon daemon)
```

Comment:
This class is a user control so it doesn't have access to a service provider of its own. Removing the usage of the _ServiceProvider.GlobalProvider_ from this class means the required instances have to be supplied by the caller.
In this case, the caller is the _GeneralOptionsDialogPage_, which does have access to a specific service provider via its _Site_ property.